sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -595,8 +601,73 @@ class Analyzer(
       }
     }
 
-    // This require transformUp to replace grouping()/grouping_id() in resolved Filter/Sort
-    def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperatorsUp {
+    private def tryResolveHavingCondition(
+        a: UnresolvedHaving, havingCond: Expression, agg: LogicalPlan): LogicalPlan = {
```

Comment:
nit: we don't need the `havingCond` parameter, as we can simply get it by `a.havingCond`

Reply:
Thanks, done in 3b48e38.